File: app/src/main/java/name/monwf/customiuizer/subs/System_VibrationAmp.java
package name.monwf.customiuizer.subs;

import android.os.Bundle;
import androidx.preference.Preference;
import android.text.format.DateFormat;

import java.util.Calendar;

import android.app.TimePickerDialog;
import android.widget.TimePicker;
import name.monwf.customiuizer.SubFragment;
import name.monwf.customiuizer.prefs.PreferenceEx;
import name.monwf.customiuizer.utils.AppHelper;

public class System_VibrationAmp extends SubFragment {

	final String key = "pref_key_system_vibration_amp_period_";
	TimePickerDialog startTimePicker;
	TimePickerDialog endTimePicker;

	@Override
	public void onActivityCreated(Bundle savedInstanceState) {
		super.onActivityCreated(savedInstanceState);

		Calendar time = Calendar.getInstance();
		boolean is24 = DateFormat.is24HourFormat(getActivity());

		int start_hour = AppHelper.getIntOfAppPrefs(key + "start_hour", time.get(Calendar.HOUR_OF_DAY));
		int start_minute = AppHelper.getIntOfAppPrefs(key + "start_minute", 0);
		updateStartTime(start_hour, start_minute);
		startTimePicker = new TimePickerDialog(getActivity(), new TimePickerDialog.OnTimeSetListener() {
			@Override
			public void onTimeSet(TimePicker timePicker, int hourOfDay, int minutes) {
				AppHelper.appPrefs.edit().putInt(key + "start_hour", hourOfDay).putInt(key + "start_minute", minutes).apply();
				updateStartTime(hourOfDay, minutes);
			}
		}, start_hour, start_minute, is24);

		findPreference(key + "start").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
			@Override
			public boolean onPreferenceClick(Preference preference) {
				startTimePicker.show();
				return true;
			}
		});

		int end_hour = AppHelper.getIntOfAppPrefs(key + "end_hour", time.get(Calendar.HOUR_OF_DAY));
		int end_minute = AppHelper.getIntOfAppPrefs(key + "end_minute", 0);
		updateEndTime(end_hour, end_minute);
		endTimePicker = new TimePickerDialog(getActivity(), new TimePickerDialog.OnTimeSetListener() {
			@Override
			public void onTimeSet(TimePicker timePicker, int hourOfDay, int minutes) {
				AppHelper.appPrefs.edit().putInt(key + "end_hour", hourOfDay).putInt(key + "end_minute", minutes).apply();
				updateEndTime(hourOfDay, minutes);
			}
		}, end_hour, end_minute, is24);

		findPreference(key + "end").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
			@Override
			public boolean onPreferenceClick(Preference preference) {
				endTimePicker.show();
				return true;
			}
		});
	}

	void updateStartTime(int hr, int min) {
		Calendar time = Calendar.getInstance();
		time.set(Calendar.HOUR_OF_DAY, hr);
		time.set(Calendar.MINUTE, min);
		((PreferenceEx)findPreference(key + "start")).setCustomSummary(DateFormat.getTimeFormat(getActivity()).format(time.getTime()));
	}

	void updateEndTime(int hr, int min) {
		Calendar time = Calendar.getInstance();
		time.set(Calendar.HOUR_OF_DAY, hr);
		time.set(Calendar.MINUTE, min);
		((PreferenceEx)findPreference(key + "end")).setCustomSummary(DateFormat.getTimeFormat(getActivity()).format(time.getTime()));
	}

}

File: app/src/main/java/tv/withaibuild/customiuizer/subs/System_VibrationAmp.kt
package tv.withaibuild.customiuizer.subs

import android.app.TimePickerDialog
import android.os.Bundle
import android.text.format.DateFormat
import android.widget.TimePicker
import androidx.preference.Preference
import tv.withaibuild.customiuizer.SubFragment
import tv.withaibuild.customiuizer.prefs.PreferenceEx
import tv.withaibuild.customiuizer.utils.AppHelper
import java.util.Calendar

class System_VibrationAmp : SubFragment() {

    private val key = "pref_key_system_vibration_amp_period_"
    private var startTimePicker: TimePickerDialog? = null
    private var endTimePicker: TimePickerDialog? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        val time = Calendar.getInstance()
        val is24 = DateFormat.is24HourFormat(activity)

        val startHour = AppHelper.getIntOfAppPrefs(key + "start_hour", time.get(Calendar.HOUR_OF_DAY))
        val startMinute = AppHelper.getIntOfAppPrefs(key + "start_minute", 0)
        updateStartTime(startHour, startMinute)
        startTimePicker = TimePickerDialog(activity, { _, hourOfDay, minutes ->
            AppHelper.appPrefs.edit().putInt(key + "start_hour", hourOfDay).putInt(key + "start_minute", minutes).apply()
            updateStartTime(hourOfDay, minutes)
        }, startHour, startMinute, is24)

        findPreference<Preference>(key + "start")?.setOnPreferenceClickListener {
            startTimePicker?.show()
            true
        }

        val endHour = AppHelper.getIntOfAppPrefs(key + "end_hour", time.get(Calendar.HOUR_OF_DAY))
        val endMinute = AppHelper.getIntOfAppPrefs(key + "end_minute", 0)
        updateEndTime(endHour, endMinute)
        endTimePicker = TimePickerDialog(activity, { _, hourOfDay, minutes ->
            AppHelper.appPrefs.edit().putInt(key + "end_hour", hourOfDay).putInt(key + "end_minute", minutes).apply()
            updateEndTime(hourOfDay, minutes)
        }, endHour, endMinute, is24)

        findPreference<Preference>(key + "end")?.setOnPreferenceClickListener {
            endTimePicker?.show()
            true
        }
    }

    private fun updateStartTime(hr: Int, min: Int) {
        val time = Calendar.getInstance().apply {
            set(Calendar.HOUR_OF_DAY, hr)
            set(Calendar.MINUTE, min)
        }
        findPreference<PreferenceEx>(key + "start")?.setCustomSummary(DateFormat.getTimeFormat(activity).format(time.time))
    }

    private fun updateEndTime(hr: Int, min: Int) {
        val time = Calendar.getInstance().apply {
            set(Calendar.HOUR_OF_DAY, hr)
            set(Calendar.MINUTE, min)
        }
        findPreference<PreferenceEx>(key + "end")?.setCustomSummary(DateFormat.getTimeFormat(activity).format(time.time))
    }
}
